scvi/inference/posterior.py

```diff
@@ -101,35 +119,73 @@ class Posterior:
     def __init__(
         self,
         model,
-        gene_dataset: GeneExpressionDataset,
+        adata: anndata.AnnData,
         shuffle=False,
         indices=None,
         use_cuda=True,
+        batch_size=128,
         data_loader_kwargs=dict(),
     ):
         self.model = model
-        self.gene_dataset = gene_dataset
+        assert "scvi_data_registry" in adata.uns.keys(), ValueError(
+            "Please run setup_anndata() on your anndata object first."
+        )
+        for key in self._data_and_attributes.keys():
+            assert key in adata.uns["scvi_data_registry"].keys(), ValueError(
+                "{} required for model but not included when setup_anndata was run".format(
+                    key
+                )
+            )
+        self.gene_dataset = BioDataset(adata, getitem_tensors=self._data_and_attributes)
```

Comment:
call this adata

Reply:
going to keep it as gene_dataset here since BioDataset is a wrapper around adata. There were `self.gene_dataset` in the trainer classes though which I've renamed to `self.adata` since those are always anndata objects.